pkg/operator/csi/csidrivercontrollerservicecontroller/csi_driver_controller_service_controller.go

```diff
@@ -119,7 +138,18 @@ func (c *CSIDriverControllerServiceController) sync(ctx context.Context, syncCon
 		return nil
 	}
 
-	manifest := replacePlaceholders(c.manifest, opSpec)
+	var clusterID string
+	if c.configInformers != nil {
+		infra, err := c.configInformers.Config().V1().Infrastructures().Lister().Get(infraConfigName)
+		if err != nil {
+			return err
+		}
+		if infra.Status.PlatformStatus != nil {
```

Comment:
this is correct? InfraName depends on another field? This at least needs a comment here. Looks like a bug now.

Reply:
Yup, we've seen this on baremetal. Added a comment

Reply:
this does not answer my question when reading this code. Why does `PlatformStatus != nil` influence why we care about cluster id?

Reply:
Sorry, I fat-fingered this.

Now that I look at it, this check doesn't make sense. I can just use `.Status.InfrastructureName` directly.